tests/background.py

```diff
@@ -361,7 +361,76 @@ def testCFHT_oldAPI(self):
         if display:
             ds9.mtv(mi, frame = 1)
 
-            
+    def testXY0(self):
+        """Test fitting the background to an image with nonzero xy0
+
+        The statsImage and background image should not vary with xy0
+        """
+        bgImageList = [] # list of background images, one per xy0
+        statsImageList = [] # list of stats images, one per xy0
+        for xy0 in (afwGeom.Point2I(0, 0), afwGeom.Point2I(1000, 500)):
+            afwdataDir = eups.productDir("afwdata")
+            if not afwdataDir:
+                print >> sys.stderr, "Skipping testCFHT as afwdata is not setup"
+                return
+
+            mi = afwImage.MaskedImageF(os.path.join(afwdataDir,
+                                                    "CFHT", "D4", "cal-53535-i-797722_1.fits"))
+            bbox = afwGeom.Box2I(afwGeom.Point2I(500, 2000), afwGeom.Point2I(2079, 4609))
+            mi = mi.Factory(mi, bbox, afwImage.LOCAL)
+            mi.setXY0(xy0)
+            self.assertEqual(mi.getBBox(), afwGeom.Box2I(xy0, bbox.getDimensions()))
+
+            bctrl = afwMath.BackgroundControl(mi.getWidth()//128, mi.getHeight()//128)
+            bctrl.getStatisticsControl().setNumSigmaClip(3.0)  
+            bctrl.getStatisticsControl().setNumIter(2)
```

Comment:
"Magic numbers" that appear in the code make me nervous -- are they just there because you need to provide some sort of argument to make things run, or do they have some significance? In this case, I think it's fairly clear that the answer is the former, but it would still be nice to mention that and/or explain why they were chosen.